src/praisonai-code/praisonai_code/cli/commands/upgrade.py

```
@@ -0,0 +1,105 @@
+            f"Automatic upgrade is not supported for a '{info.manager}' install.\n"
+            "Upgrade manually with: pip install --upgrade praisonai"
+        )
+        raise typer.Exit(1)
+
+    if not output.is_json_mode:
+        output.print(
+            f"Upgrading PraisonAI via {info.manager}: {' '.join(info.upgrade_cmd)}"
+        )
+
+    try:
+        result = subprocess.run(info.upgrade_cmd)
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)